packages/workspace/src/core/shared-interfaces.ts

```diff
@@ -41,3 +42,42 @@ export interface Environment {
   workspaceJson: any;
   workspaceResults: WorkspaceResults;
 }
+
+export interface NxWorkspaceJsonProjectDefinition<P = string> {
+  root: P;
+  sourceRoot: P;
+  projectType: 'library' | 'application';
+
+  schematic: EmptyMap | AnyStringMap;
+  architect: Record<
+    string,
+    import('@angular-devkit/core').workspaces.TargetDefinition
+  >;
+}
+
+export interface NxWorkspaceJson<P = string> {
```

Comment:
this generic is needed for schematics rules which use `join` and friends from `angular/devkit` - they don't use primitive `string` rather branded(nominal) `Path` ( which is string in reality), this properly covers these cases so API's used for schematics have it type as `Path`